src/plugins/encryption/CBC_Cipher.h

```diff
@@ -0,0 +1,124 @@
+//
+// Created by moorts on 28.10.21.
+//
+
+#ifndef SRC_CBC_CIPHER_H
+#define SRC_CBC_CIPHER_H
+
+#include "Cipher.h"
+#include "BLOCK.h"
+#include <cryptopp/cryptlib.h>
+#include <cryptopp/rijndael.h>
+#include <cryptopp/serpent.h>
+#include <cryptopp/modes.h>
+#include <cryptopp/osrng.h>
+#include <cryptopp/files.h>
+#include <cryptopp/hex.h>
+
+class CBC_Cipher : public Cipher {
+public:
+    CBC_Cipher(BLOCK b) {
+        this->b = b;
+    }
+
+    std::string encrypt(std::string plain, SecByteBlock key) const {
```

Comment:
Why all the code in the header file instead of the cpp file? (same goes for the decrypt method)